app/models/vim_performance_daily.rb

```diff
@@ -34,35 +34,20 @@ def self.find(cnt, *args)
       .to_a
   end
 
-  def self.default_time_profile(ext_options)
-    if (tz = Metric::Helper.get_time_zone(ext_options))
-      # Determine if this search falls into an existing valid TimeProfile
-      TimeProfile.rollup_daily_metrics.find_all_with_entire_tz.detect { |p| p.tz_or_default == tz }
-    end
-  end
-
   def self.find_entries(ext_options)
     ext_options ||= {}
-    ext_options[:time_profile] ||= default_time_profile(ext_options)
-
-    find_by_time_profile(ext_options)
+    find_by_time_profile(ext_options[:class], ext_options[:time_profile], ext_options[:tz])
   end
 
-  def self.find_by_time_profile(ext_options)
-    klass = ext_options[:class] || MetricRollup
-
-    # Support for multi-region DB. We need to try to find a time profile in each
-    # region that matches the selected profile to ensure that we get results for
-    # all the regions in the database. We only want one match from each region
-    # otherwise we'll end up with duplicate daily rows.
-    if (tp = ext_options[:time_profile]) && tp.rollup_daily_metrics
-      tps = TimeProfile.rollup_daily_metrics.select { |p| p.profile == tp.profile }.group_by(&:region_id).values.flatten
-      tp_ids = tps.collect(&:id)
+  # @param klass [Class] class for metrics (default: MetricRollup)
+  # @param tp [TimeProfile]
+  # @param tz [Timezone] (default: DEFUULTf)
+  def self.find_by_time_profile(klass, time_profile, tz = nil)
```

Comment:
1. I did not feel comfortable changing the signature of `find_entries`
2. I was hoping that people would use this more like a scope, and we got rid `ext_options`.

I Guess I was thinking that if we can get rid of `ext_options` and `find`, then that will make sure we are using associations and not using find or legacy accessors.